examples/puppetmaster/vagrant/windows2016/README.md

```diff
@@ -0,0 +1,190 @@
+[puppetmaster example](../../docker-compose.yml) in this repository, which
+provides a Puppet server running in a Docker container.
+
+### Overview
+
+- Mac/Linux host with Vagrant and VirtualBox installed
+- Puppet server running locally via docker-compose (using examples/puppetmaster/smoketest_e2e.sh)
+- "Vanilla" Windows2016 VM created via Vagrant
+- Installing/Running Puppet Agent
+    - Agent installation and run can be done remotely from the Mac via `vagrant powershell -c <command>`.
+    - Puppet Agent can be installed/replaced dynamically, so devs should be able to keep a long-standing
+      "vanilla" Windows VM (powered down when not in use), and install/replace Puppet agent as needed.
+
+#### Puppet and Conjur Servers
+
+- The setup is based on examples/puppetmaster/smoketest_e2e.sh, modified to:
+   - Don't do cleanup after testing Linux agents
+   - Expose Conjur on host port 30301
+   - Expose Puppet on host port 30302
+   - In Puppet server container, add /etc/hosts entry `<docker-gateway-ip> conjur`
+
+#### Puppet Agent
+
+- Puppet agent is dynamically installed/replaced on Windows VM
+- Agent version is selectable
+- Pre-provisioned machine identity using regedit and windows credentials manager
+- Follow instructions below to install/run Puppet agent
+
+## Setup
+
+### Setup Requirements
+
+This module requires that you have:
+- Oracle VirtualBox, Version 6.0 or later (installation instructions
+  [here](https://docs.oracle.com/en/virtualization/virtualbox/6.1/user/installation.html))
+- Vagrant, Version 2.2.9 or later (installation instructions
+  [here](https://www.vagrantup.com/docs/installation))
+- 60 GB of free disk space on your development server
+
+### Clone this Repository
+
+Make a local copy of this repository as follows:
+
+```sh-session
+    git clone https://github.com/cyberark/conjur-puppet
+    cd examples/puppetmaster/vagrant/windows2016_puppet
+```
+
+### Setting up Vagrant
+
+#### Install the Vagrant Reload Provisioner Plugin
+
+The [Vagrant Reload Provisioner](#https://github.com/aidanns/vagrant-reload)
+plugin can be used to conveniently reload a VM after a provisioning step
+has been done that requires a system reboot.
+
+To install the Vagrant Reload Provisioner plugin:
+
+```sh-session
+    vagrant plugin install vagrant-reload
+```
+
+### Setting up VirtualBox
+
+#### Download VirtualBox Extension Pack for Your Version of VirtualBox
+
+If you haven't done so already, download the version of VirtualBox Extension
+Pack that matches your version of VirtualBox. The appropriate versions of
+VirtualBox Extension Pack for each version of VirtualBox can be found
+[here](https://www.virtualbox.org/wiki/Downloads).
+
+## Create a Puppet Master
+
+We'll need a Puppet master to which our Puppet agent can connect.
+To do this, run the Puppet master smoketest, but leave the Puppet master
+and Conjur containers running:
+
+```sh-session
+    cd ../..
+    export CLEANUP_ON_EXIT=true
```

Comment:
I think this needs to be false

Reply:
Also seems to be `CLEAN_UP_ON_EXIT` not `CLEANUP_ON_EXIT`

Reply:
Fixed.